src/lsm/lsm_cursor.c

```diff
@@ -1574,8 +1579,14 @@ __clsm_update(WT_CURSOR *cursor)
 	WT_ERR(__cursor_needvalue(cursor));
 	WT_ERR(__clsm_enter(clsm, false, true));
 
-	if (!F_ISSET(cursor, WT_CURSTD_OVERWRITE))
+	if (!F_ISSET(cursor, WT_CURSTD_OVERWRITE)) {
 		WT_ERR(__clsm_lookup(clsm, &value));
+		/*
```

Comment:
I thought about pushing the "save a key" semantic into `__clsm_lookup`, but that seemed more confusing than this change. I'm happy to switch to that approach if it's better than the duplication in this change.